ir/type.cpp

```diff
@@ -622,6 +630,30 @@ StateValue AggregateType::extract(const StateValue &val,
                             (val.non_poison << idx).extract(h, l) });
 }
 
+StateValue AggregateType::update(const StateValue &val,
+                                 const StateValue &n,
+                                 const expr &index) const {
+  auto &elementTy = *children[0];
+  for (unsigned i = 1; i < elements; ++i) {
+    assert(elementTy.bits() == children[i]->bits());
+  }
+
+  unsigned bw_elem = elementTy.bits();
+  unsigned bw_val = val.value.bits();
+  assert(bw_val == val.non_poison.bits());
+
+  expr shift = (expr::mkUInt(elements - 1, bw_val) - index.zextOrTrunc(bw_val)) *
+    expr::mkUInt(bw_elem, bw_val);
+  expr mask = ~(expr::mkInt(-1, bw_elem).zext(bw_val - bw_elem) << shift);
+
+  StateValue n_bv = elementTy.toBV(n);
+  expr nv_shifted = (n_bv.value.zext(bw_val - bw_elem) << shift);
```

Comment:
no () needed around the whole expression

Reply:
fixed:)